### ISSUE TITLE
a11y audit: Dropdown a11y fixes

### ISSUE THREAD

**Issue by marijohannessen** [carbon-design-system/carbon-components-react#609]
### Detailed description

- [x] A WAI-ARIA widget must have an accessible name (see message help text for details). *For both the `bx--dropdown` element and the `bx--dropdown-list` element
![screen shot 2018-02-14 at 10 59 12 am](https://user-images.githubusercontent.com/5447411/36217070-206106f2-1176-11e8-90c8-d614a7358a53.png)
![screen shot 2018-02-14 at 10 59 34 am](https://user-images.githubusercontent.com/5447411/36217094-357dd0f6-1176-11e8-8c35-4b5f3a78ffa5.png)

- [x] Certain ARIA roles must contain particular children
![screen shot 2018-02-14 at 11 01 47 am](https://user-images.githubusercontent.com/5447411/36217230-7f74e744-1176-11e8-9b25-5edb5e0bd5b2.png)

Note that these are issues with the `Dropdown` component and not the `DropdownV2` component.

**Comment by tw15egan** [comment]
Should be taken care of with https://github.com/carbon-design-system/carbon-components-react/pull/675
